packages/phoenix-evals/src/phoenix/evals/core/evaluators.py

```diff
@@ -0,0 +1,503 @@
+import asyncio
+import inspect
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from functools import wraps
+from string import Formatter
+from textwrap import dedent
+from typing import Any, Callable, Dict, List, Literal, Optional, Set, Tuple, Union
+
+from typing_extensions import Mapping
+
+from phoenix.evals.llm import LLM, AsyncLLM
+
+# --- Type Aliases ---
+EvalInput = Mapping[str, Any]
+Schema = Optional[Dict[str, Any]]
+SourceType = Literal["human", "llm", "heuristic"]
+DirectionType = Literal["maximize", "minimize"]
+
+ERROR_SCORE = "ERROR"
+
+
+# --- Score model ---
+@dataclass(frozen=True)
+class Score:
+    name: Optional[str] = None
+    score: Optional[Union[float, int]] = None
+    label: Optional[str] = None
+    explanation: Optional[str] = None
+    metadata: Dict[str, Any] = field(default_factory=dict)
+    source: Optional[SourceType] = None
+    direction: Optional[DirectionType] = "maximize"
+
+
+# --- Async helper ---
+def to_thread(fn: Callable[..., Any]) -> Callable[..., Any]:
+    async def wrapper(*args: Any, **kwargs: Any) -> Any:
+        loop = asyncio.get_event_loop()
+        return await loop.run_in_executor(None, fn, *args, **kwargs)
+
+    return wrapper
+
+
+class AsyncifyMixin:
```

Comment:
My personal preference is to implement these methods manually, since delegating sync code to an executor is not always interchangeable with an async implementation we should decide whether or not pushing into a thread is appropriate for each method